### PR TITLE
implement adhAssignBadgesAction in meinberlin IdeaCollection

### PR DESCRIPTION
**Pull request by der-john**
Needs + continues #2451.

This is another step towards abolishing moving columns, because the 'badges overlay' functionality of `MovingColumns` is moved to `Badge`.

Following soon (maybe in a different PR):
- [ ] switch PolicyCompass and Euth to `adhAssignBadgesAction` as well.

### CODE REVIEW

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 1]
```diff
@@ -61,7 +66,9 @@ export var resourceActionsDirective = (
         scope: {
             resourcePath: "@",
             parentPath: "=?",
+            processUrl: "@?",
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: `processUrl` is really not a good  name for this as it does not need to be a process. Proposal: `badgeRoot`.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 2]
```diff
@@ -70,12 +77,30 @@ export var resourceActionsDirective = (
...
+                }
+            });
+            adhPermissions.bindScope(scope, () => badgeAssignmentPoolPath, "badgeAssignmentPoolOptions");
+            var params = {
+                depth: 4,
+                content_type: SIBadge.nick
+            };
+            adhHttp.get(scope.processUrl, params).then((response) => {
+                scope.badgesExist = response.data[SIPool.nick].count > 0;
+            });
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: This (expensive) code is executed even if `scope.assignBadges` is falsy. I would prefer if this was inside of `adhBadgesActionDirective`.

**Comment by xi** [comment]
Looks fine except for the comments. I am not completely sure about `parentPath`, but I would like to look into that separately.

File: src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/ProposalDetailColumn.html [hunk 3]
```diff
@@ -37,11 +37,4 @@
             </adh-meinberlin-proposal-detail>
         </adh-recompile-on-change>
     </div>
-
-    <adh-report-abuse
-        data-ng-switch-when="overlays"
-        data-ng-if="overlay === 'abuse'"
-        data-modals="column"
-        data-url="{{ proposalUrl | adhParentPath | adhResourceUrl | adhCanonicalUrl }}">
-    </adh-report-abuse>
```
Comment (der-john) [inline src/meinberlin/meinberlin/static/js/Packages/Meinberlin/IdeaCollection/ProposalDetailColumn.html]: thanks!

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 2 above)
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Got it.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 1 above)
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: I renamed it to `resourceWithBadgesUrl`.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts (hunk 2 above)
Comment (der-john) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: > This (expensive) code is executed even if scope.assignBadges is falsy. I would prefer if this was inside of adhBadgesActionDirective.

I added a check
```
if (scope.assignBadges) { ... }
```
around this logic. The parameters are needed in `ResourceActions.html`.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts [hunk 4]
```diff
@@ -90,22 +90,24 @@ export var resourceActionsDirective = (
...
+                });
+                adhPermissions.bindScope(scope, () => badgeAssignmentPoolPath, "badgeAssignmentPoolOptions");
+                var params = {
+                    depth: 4,
+                    content_type: SIBadge.nick
+                };
+                adhHttp.get(scope.resourceWithBadgesUrl, params).then((response) => {
+                    scope.badgesExist = response.data[SIPool.nick].count > 0;
+                });
+            }
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: This does not work for the case that `scope.assignBadges` switches from false to true after linking. I really thing we should move this logic into the action itself.

Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts]: Fixed in 	94c133d

**Pull request merged**